conda/core/subdir_data.py

```diff
@@ -3,59 +3,76 @@
 
 from __future__ import annotations
 
-import hashlib
 import json
-import pathlib
 import pickle
 import re
 import warnings
 from collections import UserList, defaultdict
 from contextlib import closing
 from errno import EACCES, ENODEV, EPERM, EROFS
 from functools import partial
-from io import open as io_open
 from itertools import chain, islice
 from logging import getLogger
 from mmap import ACCESS_READ, mmap
 from os.path import dirname, exists, isdir, join, splitext
+from pathlib import Path
 from time import time
 
 from genericpath import getmtime, isfile
 
 from conda.common.iterators import groupby_to_dict as groupby
 from conda.gateways.repodata import (
     CondaRepoInterface,
+    RepodataCache,
     RepodataIsEmpty,
+    RepodataOnDisk,
+    RepodataState,
     RepoInterface,
     Response304ContentUnchanged,
+    cache_fn_url,
 )
 
 from .. import CondaError
 from .._vendor.boltons.setutils import IndexedSet
 from ..auxlib.ish import dals
 from ..base.constants import CONDA_PACKAGE_EXTENSION_V1, REPODATA_FN
 from ..base.context import context
-from ..common.compat import ensure_binary, ensure_unicode
+from ..common.compat import ensure_unicode
 from ..common.io import DummyExecutor, ThreadLimitedThreadPoolExecutor, dashlist
 from ..common.path import url_to_path
 from ..common.url import join_url
 from ..core.package_cache_data import PackageCacheData
+from ..deprecations import deprecated
 from ..exceptions import CondaUpgradeError, NotWritableError, UnavailableInvalidChannel
 from ..gateways.disk import mkdir_p, mkdir_p_sudo_safe
 from ..gateways.disk.delete import rm_rf
-from ..gateways.disk.update import touch
 from ..models.channel import Channel, all_channel_urls
 from ..models.match_spec import MatchSpec
 from ..models.records import PackageRecord
 from ..trust.signature_verification import signature_verification
 
 log = getLogger(__name__)
 
-REPODATA_PICKLE_VERSION = 31
+REPODATA_PICKLE_VERSION = 30
 MAX_REPODATA_VERSION = 1
 REPODATA_HEADER_RE = b'"(_etag|_mod|_cache_control)":[ ]?"(.*?[^\\\\])"[,}\\s]'  # NOQA
 
 
+def get_repo_interface() -> type[RepoInterface]:
+    if "jlap" in context.experimental:
+        try:
+            from conda.gateways.repodata.jlap.interface import JlapRepoInterface
+
+            return JlapRepoInterface
+        except ImportError as e:  # pragma: no cover
+            warnings.warn(
+                "Could not load the configured jlap repo interface. "
+                f"Is the required jsonpatch package installed?  {e}"
+            )
```

Comment:
I'm thinking we should mark `jsonpatch` as a dependency (in the recipe and in `requirements.txt`) instead of including this extra try-except.

Reply:
Maybe. How experimental is the feature? jsonpatch is 14kb and its dependency jsonpointer is 11kb.

Reply:
I think I agree it makes sense to just depend on jsonpatch all the time. Added to https://github.com/conda/conda/pull/12461 which is intended to be the next step after this `features/repodata-jlap`